pysat/tests/test_instruments.py

```diff
@@ -34,70 +33,32 @@
 instruments = generate_instrument_list(inst_loc=pysat.instruments,
                                        user_info=user_info)
 
-# The following lines apply the custom instrument lists to each type of test
-method_list = [func for func in dir(InstTestClass)
-               if callable(getattr(InstTestClass, func))]
-# Search tests for iteration via pytestmark, update instrument list
-for method in method_list:
-    if hasattr(getattr(InstTestClass, method), 'pytestmark'):
-        # Get list of names of pytestmarks
-        Nargs = len(getattr(InstTestClass, method).pytestmark)
-        names = [getattr(InstTestClass, method).pytestmark[j].name
-                 for j in range(0, Nargs)]
-        # Add instruments from your library
-        if 'all_inst' in names:
-            mark = pytest.mark.parametrize("inst_name", instruments['names'])
-            getattr(InstTestClass, method).pytestmark.append(mark)
-        elif 'download' in names:
-            mark = pytest.mark.parametrize("inst_dict", instruments['download'])
-            getattr(InstTestClass, method).pytestmark.append(mark)
-        elif 'no_download' in names:
-            mark = pytest.mark.parametrize("inst_dict",
-                                           instruments['no_download'])
-            getattr(InstTestClass, method).pytestmark.append(mark)
-
 
 class TestInstruments(InstTestClass):
     """Main class for instrument tests.
 
     Note
     ----
-    Uses class level setup and teardown so that all tests use the same
-    temporary directory. We do not want to geneate a new tempdir for each test,
-    as the load tests need to be the same as the download tests.
+    All standard tests, setup, and teardown inherited from the core pysat
+    instrument test class.
 
     """
 
-    def setup_class(self):
-        """Initialize the testing setup once before all tests are run."""
-        # Make sure to use a temporary directory so that the user's setup is not
-        # altered
-        self.tempdir = tempfile.TemporaryDirectory()
-        self.saved_path = pysat.params['data_dirs']
-        pysat.params['data_dirs'] = self.tempdir.name
-        # Developers for instrument libraries should update the following line
-        # to point to their own subpackage location, e.g.,
-        # self.inst_loc = mypackage.instruments
-        self.inst_loc = pysat.instruments
-        return
-
-    def teardown_class(self):
-        """Clean up downloaded files and parameters from tests."""
-        pysat.params['data_dirs'] = self.saved_path
-        self.tempdir.cleanup()
-        del self.inst_loc, self.saved_path, self.tempdir
-        return
-
     # Custom package unit tests can be added here
 
-    # Custom Integration Tests added to all instruments.
+    # Custom Integration Tests added to all test instruments in core package.
     @pytest.mark.parametrize("inst_dict", [x for x in instruments['download']])
```

Comment:
```suggestion
    @pytest.mark.parametrize("inst_dict", InstTestClass.[x for x in instruments['download']])
```
Conversely, we could attach the instrument list to the test class directly and then invoke them in pytest like this.

Reply:
Thinking about this from how I would explain the implementation to a new user, I think this is the best option, but open to feedback.

Reply:
I am fine either way, I keep going back and forth between liking one over the other.  I am curious as to whether you need to create a list from an existing list, though.